src/memoize.ts

```diff
@@ -18,19 +18,20 @@
  *     results. It has all the arguments applied to it and must be synchronous.
  * @returns a memoized version of fn
  */
-export function memoize<FnType extends Function>(
+// tslint:disable:no-any defining it this way is more precise than Function so is still preferable
+export function memoize<FnType extends (...args: any[]) => Promise<any>>(
   fn: FnType,
-  // tslint:disable-next-line:no-any (w/o type for Function args, can't assert a type here)
-  hasher: Function = (arg: any) => arg,
+  // tslint:disable:no-any hasher can return any value that can be used as a map key
+  hasher: (...args: Parameters<FnType>) => any = (...args) => args[0],
   timeoutMs?: number,
 ): FnType & { reset: FnType; clear: () => void } {
```

Comment:
think this is actually a `(...args: Parameters<FnType>) => void` (not a promise so the signature is actually wrong for consumers rn)